packages/http-client-csharp/generator/Microsoft.Generator.CSharp/src/Providers/PropertyProvider.cs

```diff
@@ -28,6 +28,12 @@ public class PropertyProvider
         public PropertyProvider(InputModelProperty inputProperty)
         {
             var propertyType = CodeModelPlugin.Instance.TypeFactory.CreateCSharpType(inputProperty.Type);
+            // We represent property being optional by making it nullable (when it is a value type)
+            // Except in the case of collection where there is a special handling
+            if (inputProperty is { IsRequired: false } && inputProperty.Type is not InputNullableType && !propertyType.IsCollection)
+            {
+                propertyType = propertyType.WithNullable(true);
+            }
```

Comment:
I think the condition here is not so accurate.

`CreateCSharpType` already handles the case of nullable therefore we should not be checking this here.
I think it should enough by just checking
```csharp
!inputProperty.IsRequired && propertyType is {IsCollection: false, IsValueType: true}
```
Or it actually also makes sense if we just check `!inputProperty.IsRequired` - whether it serialized as nullable is handled in other places, not in the CSharpType, it does not harm (and it is true) when it is optional, the reference type is also nullable.